packages/java/parser-jvm-plugin-nonnull/src/main/java/com/vaadin/fusion/parser/plugins/nonnull/NonnullPlugin.java

```diff
@@ -0,0 +1,59 @@
+package com.vaadin.fusion.parser.plugins.nonnull;
+
+import java.util.Collection;
+
+import javax.annotation.Nonnull;
+
+import com.vaadin.fusion.parser.core.Plugin;
+import com.vaadin.fusion.parser.core.PluginConfiguration;
+import com.vaadin.fusion.parser.core.RelativeClassInfo;
+import com.vaadin.fusion.parser.core.SharedStorage;
+import com.vaadin.fusion.parser.plugins.backbone.AssociationMap;
+import com.vaadin.fusion.parser.plugins.backbone.BackbonePlugin;
+import com.vaadin.fusion.parser.utils.PluginException;
+
+public final class NonnullPlugin implements Plugin {
+    private Collection<String> annotations;
+    private int order = 100;
+
+    @Override
+    public void execute(@Nonnull Collection<RelativeClassInfo> endpoints,
+            @Nonnull Collection<RelativeClassInfo> entities,
+            @Nonnull SharedStorage storage) {
+        var associationMap = (AssociationMap) storage.getPluginStorage()
```

Comment:
No, because `PluginStorage` is a core API, while `AssociationMap` is a part of `BackbonePlugin`. Basically, `BackbonePlugin`, as a regular plugin, adds to the Storage something that Storage is not aware of.

Reply:
Per discussion, this can be solved in a separate PR, made a ticket https://github.com/vaadin/fusion/issues/180.